docs/proposed/http-introducer.rst

```diff
@@ -0,0 +1,153 @@
+By configuring your client with one small static connection hint,
+the "HTTP Introducer" system enables the client to find more information and future updates to that information.
+
+Before attempting this configuration,
+you should have an HTTP-style *introducer fURL* from the operator of an HTTP Introducer-enabled Tahoe-LAFS storage grid.
+
+The HTTP Introducer is configured in two parts.
+First,
+the *introducer fURL* is written to a new file.
+This configuration should be treated as a secret.
+The node's ``private`` directory is a good location to consider for this file.
+For example,
+for a grid you refer to as the "foo grid" you might use ``private/foogrid-introducer``.
+
+Next,
+in ``tahoe.cfg`` the ``grid-introducer-path`` item in the ``[client]`` section is set to refer to this file.
+For example::
+
+  [client]
+  http-introducer-path = private/foogrid-introduction
+
+Start a Tahoe-LAFS client node with these items configured and the client will be able to find and follow all storage servers that are part of that grid.
+
+Use As A Storage Provider
+-------------------------
+
+You want your Tahoe-LAFS storage node to be able to publish its connection details where clients can find it.
+A storage server can publish its storage announcement to the HTTP Introducer.
+The HTTP Introducer is then responsible for delivering it to interested parties.
+
+The configuration for a storage node is exactly the same as the configuration for a client node,
+as described above.
+
+After the storage server has been configured this way and started it will publish its announcement to the HTTP Introducer.
+
+Use As A Grid Coordinator
+-------------------------
+
+Setup
+~~~~~
+
+You want to offer a collection of storage servers as a Tahoe-LAFS storage grid.
+
+The ``http-introducer`` tool is a stand-alone server which implements basic publish/subscribe functionality.
+
+The first step is to create the introducer's persistent state::
+
+  http-introducer create --config <path> --listen-endpoint <endpoint> [--certificate-path <path> --private-key-path <path>]
+
+This is an error if ``<path>`` exists already.
+If it does not then a new HTTP Introducer configuration is written there.
+If ``<path>`` is ``-`` then the configuration is written to stdout.
+
+``--listen-endpoint`` may be used repeatedly to listen on multiple addresses.
+Whatever endpoint type is chosen,
+``http-introducer`` will *always* _automatically_ negotiate TLS over it.
+
+If given,
+the files given by ``--certificate-path`` and ``--private-key-path`` have their contents read and added to the configuration state.
+After the execution of this command,
+these files are not read again.
+If not given,
+a new private key and self-signed certificate is generated and used.
+
+This is the HTTP Introducer's state and is required by future commands.
+It is the operator's responsibility to persist this state.
```

Comment:
Might be worth repeating here that this state is Seekrit Information and to treat accordingly (even if that's already strongly implied with `--private-key` etc etc) and the consequences (i.e. that you can forever impersonate this introducer and direct clients to malicious servers).